catalogue/src/main/kotlin/com/tidal/sdk/catalogue/generated/infrastructure/URLAdapter.kt

```diff
@@ -0,0 +1,21 @@
+package com.tidal.sdk.catalogue.generated.infrastructure
+
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializer
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.SerialDescriptor
+import java.net.URL
+
+@Serializer(forClass = URL::class)
+object URLAdapter : KSerializer<URL> {
```

Comment:
I'd argue that we should not use URLs in data classes since it's quite a heavy object and does more than you'd think (i.e. validation of schema, supports networking, accidentally uses internet etc).

I'd probably keep them as string in the data class representation until the user actually needs the URL to open a connection (which is way less often than the data class got instantiated)

Reply:
this is generated code - I'll have a look if that can be changed easily.

Reply:
Yeah, I understood this very late - sorry for all the comments